File: crates/cs/csharp/ZeroDDS.Cdr/src/XcdrException.cs
// SPDX-License-Identifier: Apache-2.0
// Copyright 2026 ZeroDDS Contributors
//
// Runtime error for the XCDR2 encoder/decoder.

using System;

namespace ZeroDDS.Cdr;

/// <summary>
/// Thrown by the XCDR2 encoder/decoder on wire-format errors
/// (z.B. Bounds-Verletzung, ungueltiger DHEADER, unbekanntes LC).
/// </summary>
public sealed class XcdrException : Exception
{
    /// <summary>Default-Constructor.</summary>
    public XcdrException() : base() { }

    /// <summary>Constructor with an error message.</summary>
    public XcdrException(string message) : base(message) { }

    /// <summary>Constructor with an error message + inner exception.</summary>
    public XcdrException(string message, Exception inner) : base(message, inner) { }
}

File: crates/java-omgdds/java/src/main/java/org/zerodds/cdr/XcdrException.java
// SPDX-License-Identifier: Apache-2.0
package org.zerodds.cdr;

/**
 * Runtime error in the XCDR2 encoder/decoder.
 *
 * <p>Thrown on bounds violations, invalid wire data,
 * decoder stream underflow and encoder range violations. Extends
 * {@link RuntimeException} because the DDS Java PSM does not use checked exceptions
 * an den Sample-API-Pfaden vorsieht.
 */
public final class XcdrException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    /** Constructs with a detail message. */
    public XcdrException(String message) {
        super(message);
    }

    /** Constructs with a detail message and cause. */
    public XcdrException(String message, Throwable cause) {
        super(message, cause);
    }
}
